{
  "abstract": [
    {
      "text": "An array containing the current call stack symbols.",
      "type": "text"
    }
  ],
  "documentVersion": 0,
  "hierarchy": {
    "paths": [
      [
        "doc://com.apple.documentation/documentation/technologies",
        "doc://com.apple.documentation/documentation/foundation",
        "doc://com.apple.documentation/documentation/foundation/errors_and_exceptions",
        "doc://com.apple.documentation/documentation/foundation/nsexception"
      ]
    ]
  },
  "identifier": {
    "interfaceLanguage": "swift",
    "url": "doc://com.apple.documentation/documentation/foundation/nsexception/1416845-callstacksymbols"
  },
  "kind": "symbol",
  "legacy_identifier": 1416845,
  "legalNotices": {
    "copyright": "Copyright &copy; 2025 Apple Inc. All rights reserved.",
    "privacyPolicy": "https://www.apple.com/privacy/privacy-policy",
    "termsOfUse": "https://www.apple.com/legal/internet-services/terms/site.html"
  },
  "metadata": {
    "externalID": "c:objc(cs)NSException(py)callStackSymbols",
    "fragments": [
      {
        "kind": "text",
        "text": "var "
      },
      {
        "kind": "identifier",
        "text": "callStackSymbols"
      },
      {
        "kind": "text",
        "text": ": [String]"
      }
    ],
    "modules": [
      {
        "name": "Foundation"
      }
    ],
    "parent": {
      "title": "NSException"
    },
    "platforms": [
      {
        "current": "18.4",
        "introducedAt": "4.0",
        "name": "iOS"
      },
      {
        "current": "18.4",
        "introducedAt": "4.0",
        "name": "iPadOS"
      },
      {
        "current": "18.4",
        "introducedAt": "13.1",
        "name": "Mac Catalyst"
      },
      {
        "current": "15.4",
        "introducedAt": "10.6",
        "name": "macOS"
      },
      {
        "current": "18.4",
        "introducedAt": "9.0",
        "name": "tvOS"
      },
      {
        "current": "2.4",
        "introducedAt": "1.0",
        "name": "visionOS"
      },
      {
        "current": "11.4",
        "introducedAt": "2.0",
        "name": "watchOS"
      }
    ],
    "role": "symbol",
    "roleHeading": "Instance Property",
    "symbolKind": "instp",
    "title": "callStackSymbols"
  },
  "primaryContentSections": [
    {
      "declarations": [
        {
          "languages": [
            "swift"
          ],
          "platforms": [
            "iOS",
            "iPadOS",
            "Mac Catalyst",
            "macOS",
            "tvOS",
            "visionOS",
            "watchOS"
          ],
          "tokens": [
            {
              "kind": "keyword",
              "text": "var"
            },
            {
              "kind": "text",
              "text": " "
            },
            {
              "kind": "identifier",
              "text": "callStackSymbols"
            },
            {
              "kind": "text",
              "text": ": "
            },
            {
              "kind": "text",
              "text": "["
            },
            {
              "identifier": "doc://com.apple.Swift/documentation/Swift/String",
              "kind": "typeIdentifier",
              "preciseIdentifier": "s:SS",
              "text": "String"
            },
            {
              "kind": "text",
              "text": "]"
            },
            {
              "kind": "text",
              "text": " { "
            },
            {
              "kind": "keyword",
              "text": "get"
            },
            {
              "kind": "text",
              "text": " }"
            }
          ]
        }
      ],
      "kind": "declarations"
    },
    {
      "content": [
        {
          "anchor": "discussion",
          "level": 2,
          "text": "Discussion",
          "type": "heading"
        },
        {
          "inlineContent": [
            {
              "text": "An array of strings describing the call stack backtrace at the moment the exception was first raised. The format of each string is determined by the ",
              "type": "text"
            },
            {
              "code": "backtrace_symbols()",
              "type": "codeVoice"
            },
            {
              "text": " API",
              "type": "text"
            }
          ],
          "type": "paragraph"
        }
      ],
      "kind": "content"
    }
  ],
  "references": {
    "doc://com.apple.Swift/documentation/Swift/String": {
      "identifier": "doc://com.apple.Swift/documentation/Swift/String",
      "kind": "symbol",
      "role": "symbol",
      "title": "String",
      "type": "topic",
      "url": "/documentation/swift/string"
    },
    "doc://com.apple.documentation/documentation/foundation": {
      "identifier": "doc://com.apple.documentation/documentation/foundation",
      "kind": "symbol",
      "role": "collection",
      "title": "Foundation",
      "type": "topic",
      "url": "/documentation/foundation"
    },
    "doc://com.apple.documentation/documentation/foundation/errors_and_exceptions": {
      "identifier": "doc://com.apple.documentation/documentation/foundation/errors_and_exceptions",
      "kind": "article",
      "role": "collectionGroup",
      "title": "Errors and Exceptions",
      "type": "topic",
      "url": "/documentation/foundation/errors_and_exceptions"
    },
    "doc://com.apple.documentation/documentation/foundation/nsexception": {
      "abstract": [
        {
          "text": "An object that represents a special condition that interrupts the normal flow of program execution.",
          "type": "text"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/nsexception",
      "kind": "symbol",
      "role": "symbol",
      "title": "NSException",
      "type": "topic",
      "url": "/documentation/foundation/nsexception"
    },
    "doc://com.apple.documentation/documentation/foundation/nsexception/1412165-callstackreturnaddresses": {
      "abstract": [
        {
          "text": "The call return addresses related to a raised exception.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "callStackReturnAddresses"
        },
        {
          "kind": "text",
          "text": ": [NSNumber]"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/nsexception/1412165-callstackreturnaddresses",
      "kind": "symbol",
      "role": "symbol",
      "title": "callStackReturnAddresses",
      "type": "topic",
      "url": "/documentation/foundation/nsexception/1412165-callstackreturnaddresses"
    },
    "doc://com.apple.documentation/documentation/foundation/nsexception/1416845-callstacksymbols": {
      "abstract": [
        {
          "text": "An array containing the current call stack symbols.",
          "type": "text"
        }
      ],
      "fragments": [
        {
          "kind": "text",
          "text": "var "
        },
        {
          "kind": "identifier",
          "text": "callStackSymbols"
        },
        {
          "kind": "text",
          "text": ": [String]"
        }
      ],
      "identifier": "doc://com.apple.documentation/documentation/foundation/nsexception/1416845-callstacksymbols",
      "kind": "symbol",
      "role": "symbol",
      "title": "callStackSymbols",
      "type": "topic",
      "url": "/documentation/foundation/nsexception/1416845-callstacksymbols"
    },
    "doc://com.apple.documentation/documentation/technologies": {
      "identifier": "doc://com.apple.documentation/documentation/technologies",
      "kind": "technologies",
      "title": "Technologies",
      "type": "topic",
      "url": "/documentation/technologies"
    }
  },
  "schemaVersion": {
    "major": 0,
    "minor": 3,
    "patch": 0
  },
  "sections": [],
  "seeAlsoSections": [
    {
      "generated": true,
      "identifiers": [
        "doc://com.apple.documentation/documentation/foundation/nsexception/1412165-callstackreturnaddresses"
      ],
      "title": "Getting Exception Stack Frames"
    }
  ],
  "variants": [
    {
      "paths": [
        "documentation/foundation/nsexception/1416845-callstacksymbols"
      ],
      "traits": [
        {
          "interfaceLanguage": "occ"
        }
      ]
    },
    {
      "paths": [
        "documentation/foundation/nsexception/1416845-callstacksymbols"
      ],
      "traits": [
        {
          "interfaceLanguage": "swift"
        }
      ]
    }
  ]
}
